kanidmd/src/lib/config.rs

```diff
@@ -71,6 +71,11 @@ pub struct Configuration {
     pub cookie_key: [u8; 32],
     pub integration_test_config: Option<Box<IntegrationTestConfig>>,
     pub log_level: Option<u32>,
+    // TODO: online backup settings
```

Comment:
You could make this a section in the config by adding a struct called OnlineBackup and that will allow the config tou have a section like 

```
[online_backup]
path = 
interval = 
versions =
```

I think something to keep in mind is that if you make it an Option<OnlineBackup> then this section is optional and it's abscence means "don't run automatic backups".

Reply:
Yes, a struct might be good. There are quite a few items related to the backup.

What would you like to have? Should it be optional? The main reason for me was th be able to switch it on and off. But it could als be a separate option in the new config section.

With the latter, everything could be required, but  will require action to upgrade config files. 

Another thought. How much do we want to be actively set in the config? I think there are also some good defaults that might be used auyomatically.

That said, i am also wondering if the online backup should be running by default or not.
Once a day might not be too bad.

Reply:
I think it should be optional and not enabled by default. But I think we should show it in the example config in the book. 

But having it as a seperate section does allow it on/off by the presence /abscence of the section in the config :) 

And also remember, that within the struct for the Option<OnlineBackup> or whatever, OnlineBackup can also have values that ahve defaults within them so that just specifying "the section header" makes it present, then we get defaults for all the other values :) https://serde.rs/attr-default.html may help you because you can set the default values when not present. in this case. Does that help?

Reply:
You meant I should add something like this `#[serde(default = "default_resource")]` to the OnlineBackup struct? Got some errors about (De)Serialisation (I think is would lead to implement FromString ToString?!) ... and I am not fully sure - would this be the place where we put default options for OnlineBackup?

With the current state, you will have to uncomment the entire online_backup section, but I think this is fair, as you might want to adjust everything there. We can also change that, but then I need some more hints ;-)

Reply:
Yeah, but those `#[serde(default = "default_resource")]` are on the attributes. Something like:

```
fn default_to_keep() -> usize { 7 }

fn default_schedule() -> Cron { ... }

struct OnlineBackup {
    outpath = String,
    #[serde(default = "default_schedule")]
    schedule = Cron,
    #[serde(default = "default_to_keep")]
    number_to_keep = usize,
}
```

Does that example help?

Reply:
:facepalm:  Yes, thank you very much!

Reply:
No need for "🤦 ", I'm happy to help :)